Src/FluentAssertions/Equivalency/MemberVisibilityExtensions.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Concurrent;
+using Reflectify;
+
+namespace FluentAssertions.Equivalency;
+
+internal static class MemberVisibilityExtensions
+{
+    private static readonly ConcurrentDictionary<MemberVisibility, MemberKind> Cache = new();
+
+    public static MemberKind ToMemberKind(this MemberVisibility visibility)
+    {
+        return Cache.GetOrAdd(visibility, v =>
```

Comment:
To have the compiler help enforce that we don't capture anything
```suggestion
        return Cache.GetOrAdd(visibility, static v =>
```

Reply:
Cool. Didn't know about this one.